File: src/Entity/Notification.php
<?php

namespace App\Entity;

use App\Repository\NotificationRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: NotificationRepository::class)]
#[ORM\Table(name: 'notification')]
class Notification
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(type: 'integer')]
    #[Assert\NotNull(message: 'L\'identifiant utilisateur est obligatoire.')]
    #[Assert\Positive(message: 'L\'identifiant utilisateur doit être positif.')]
    private int $userId = 0;

    #[ORM\Column(length: 20)]
    #[Assert\NotBlank(message: 'Le type d\'utilisateur est obligatoire.')]
    #[Assert\Choice(choices: ['CLIENT','PARTENAIRE','ADMIN'], message: 'Type utilisateur invalide.')]
    private string $userType = '';

    #[ORM\Column(length: 50)]
    #[Assert\NotBlank(message: 'Le type de notification est obligatoire.')]
    private string $type = '';

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: 'Le titre est obligatoire.')]
    #[Assert\Length(max: 255, maxMessage: 'Le titre ne peut pas dépasser {{ limit }} caractères.')]
    private string $titre = '';

    #[ORM\Column(type: 'text')]
    #[Assert\NotBlank(message: 'Le message est obligatoire.')]
    private string $message = '';

    #[ORM\Column(type: 'boolean', nullable: true)]
    private ?bool $lue = false;

    #[ORM\Column(type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $dateCreation = null;

    #[ORM\Column(type: 'integer', nullable: true)]
    #[Assert\Positive(message: 'L\'identifiant activité doit être positif.')]
    private ?int $activiteId = null;

    #[ORM\Column(type: 'integer', nullable: true)]
    #[Assert\Positive(message: 'L\'identifiant demande doit être positif.')]
    private ?int $demandeId = null;

    public function getId(): ?int { return $this->id; }

    public function getUserId(): int { return $this->userId; }
    public function setUserId(int $userId): static { $this->userId = $userId; return $this; }

    public function getUserType(): string { return $this->userType; }
    public function setUserType(string $userType): static { $this->userType = $userType; return $this; }

    public function getType(): string { return $this->type; }
    public function setType(string $type): static { $this->type = $type; return $this; }

    public function getTitre(): string { return $this->titre; }
    public function setTitre(string $titre): static { $this->titre = $titre; return $this; }

    public function getMessage(): string { return $this->message; }
    public function setMessage(string $message): static { $this->message = $message; return $this; }

    public function getLue(): ?bool { return $this->lue; }
    public function setLue(?bool $lue): static { $this->lue = $lue; return $this; }

    public function getDateCreation(): ?\DateTimeInterface { return $this->dateCreation; }
    public function setDateCreation(?\DateTimeInterface $dateCreation): static { $this->dateCreation = $dateCreation; return $this; }

    public function getActiviteId(): ?int { return $this->activiteId; }
    public function setActiviteId(?int $activiteId): static { $this->activiteId = $activiteId; return $this; }

    public function getDemandeId(): ?int { return $this->demandeId; }
    public function setDemandeId(?int $demandeId): static { $this->demandeId = $demandeId; return $this; }
}

File: nexora_java/src/main/java/com/pi/entities/Notification.java
package com.pi.entities;

import java.time.LocalDateTime;

public class Notification {
    private int id;
    private int userId;
    private String userType; // CLIENT ou PARTENAIRE
    private String type; // NOUVELLE_DEMANDE, ACCEPTATION, REFUS, PAIEMENT, etc.
    private String titre;
    private String message;
    private boolean lue;
    private LocalDateTime dateCreation;
    private int activiteId;
    private int demandeId;

    public Notification() {
        this.dateCreation = LocalDateTime.now();
        this.lue = false;
    }

    public Notification(int userId, String userType, String type, String titre, String message) {
        this.userId = userId;
        this.userType = userType;
        this.type = type;
        this.titre = titre;
        this.message = message;
        this.dateCreation = LocalDateTime.now();
        this.lue = false;
    }

    // Getters et Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public String getUserType() {
        return userType;
    }

    public void setUserType(String userType) {
        this.userType = userType;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public boolean isLue() {
        return lue;
    }

    public void setLue(boolean lue) {
        this.lue = lue;
    }

    public LocalDateTime getDateCreation() {
        return dateCreation;
    }

    public void setDateCreation(LocalDateTime dateCreation) {
        this.dateCreation = dateCreation;
    }

    public int getActiviteId() {
        return activiteId;
    }

    public void setActiviteId(int activiteId) {
        this.activiteId = activiteId;
    }

    public int getDemandeId() {
        return demandeId;
    }

    public void setDemandeId(int demandeId) {
        this.demandeId = demandeId;
    }

    @Override
    public String toString() {
        return "Notification{" +
                "id=" + id +
                ", userId=" + userId +
                ", type='" + type + '\'' +
                ", titre='" + titre + '\'' +
                ", lue=" + lue +
                ", dateCreation=" + dateCreation +
                '}';
    }
}
